docs/subnets/subnet-hyperparameters.md

```diff
@@ -105,27 +106,53 @@ Consider Subnet-1 where `max_weight_limit` is set to 455 and `min_allowed_weight
 
 --- -->
 
-## immunity_period
+## commit_reveal_weights_enabled
 
 **Description**
-: The immunity period is expressed in number of blocks. The immunity period is the number of blocks given to a subnet miner or a subnet validator at a UID before they are considered available for deregistration. 
+: Enables the [Commit Reveal](./commit-reveal.md) feature.
 
 **Value**
-: Set to `7200` blocks for Subnet-1. The `immunity_period` at a UID starts when a subnet validator or a subnet miner is registered into the subnet. 
+: `True` or `False`. Enables or disables the feature, default: false.
 
 **Setting**
-: This parameter can be changed by the subnet owner. The value of this parameter varies from subnet to subnet. 
+: This parameter can be changed by the subnet owner.
+
+## commit_reveal_weights_interval
+
+**Description**
+: The interval, measured as a number of blocks, that elapses before unencrypted weights are revealed.
+
+**Value**
+: The commit reveal interval, as an integer count of blocks.
 
-A subnet miner or a subnet validator at a UID can perform poorly during the `immunity_period` without risking deregistration. If the UID still does not perform well even after the expiry of the `immunity_period`, then the subnet miner or subnet validator at that UID can be removed from the subnet. They will be removed when a new entity, i.e., a subnet miner or a subnet validator, requests to join the subnet.
+**Setting**
+:  This parameter can be changed by the subnet owner, and must be tuned carefully for the subnet. It should always be greater than the [immunity period](#immunity_period) to avoid unintended miner deregistration&mdash;see [Commit Reveal and Immunity Period](./commit-reveal.md#commit-reveal-and-immunity-period).
 
-When a subnet miner or a subnet validator is deregistered, they are required to register again to be considered for the subnet. 
+## immunity_period
+
+**Description**
+: The [Immunity Period](../glossary.md#immunity-period) within a subnet. This is a grace period, measured in blocks, granted to a miner or a validator when registered at a UID on a given subnet, that elapses before they are considered for deregistration due to performance.
+
+If the UID still does not perform well even after the expiry of the `immunity_period`, the miner or validator at that UID can be removed from the subnet when a new miner or validator requests to join.
```

Comment:
Might be good to add what happens if (somehow) *all* the peers are immune. In this case we deregister the peer with the lowest pruning score, anyway.

Another case is when all peers have the same pruning score, in which case we deregister the oldest registered peer. First we do this for the non-immune peers, but if all are immune, we consider all peers, as above.

Reply:
Might be better as a footnote. Because it's unlikely to ever occur